includes/modules/pages/checkout_confirmation/jscript_paypal_incontext.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * Javascript enabler for providing and handling the PayPal 
+ *   Express In Context shopping experience.
+ *
+ * @package page
+ * @copyright Copyright 2003-2016 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id: Author: mc12345678  Sat Aug 13 11:35:35 2016 -0400 Modified in v1.6.0 $
+ */
+ 
+  if (   defined('MODULE_PAYMENT_PAYPALWPP_CHECKOUTSTYLE') && MODULE_PAYMENT_PAYPALWPP_CHECKOUTSTYLE == 'InContext' 
+      && defined('MODULE_PAYMENT_PAYPALWPP_MERCHANTID')    && MODULE_PAYMENT_PAYPALWPP_MERCHANTID    != '' 
+      && defined('MODULE_PAYMENT_PAYPALWPP_STATUS')        && MODULE_PAYMENT_PAYPALWPP_STATUS        == 'True') {
+        
+//        require(DIR_WS_MODULES . 'payment/paypal/paypalwpp_incontext.php');
+if (isset($_SESSION['paypal_ec_token']) && !empty($_SESSION['paypal_ec_token'])
+    && isset($_SESSION['paypal_ec_markflow']) && $_SESSION['paypal_ec_markflow'] = 1) {
+
+}?>
```

Comment:
I meant to write something earlier about the change(s) provided, might help with some of the understanding.

Currently, process_button in paypalwpp.php only provides information related to in context process if there is no previous paypal_ec_token session data and the appropriate action/settings have been applied to allow PayPal in context to be used. In straight forward processing this is fine because either the customer visited PayPal earlier and established all of the necessary information to permit shipping verification and payment collection or they haven't yet and therefore at the checkout_confirmation page are ready to goto PayPal (in this discussion In Context) where they apply the necessary credentials to complete payment. The thing is, if at any point during this process they select the cancel option in the PayPal window, then they are redirected to the cancelurl which was set depending on the logged in status of the user before attempting purchase. Well, at the point of checkout_confirmation, if I understand correctly the user either as themselves or a temporary user is logged in and therefore the cancel url is back to the checkout_shipping page. But, there is nothing at the checkout_shipping page nor forward towards checkout_confirmation that modifies the paypal_ec_token other than potentially recreating one when the payment is again posted at the checkout_confirmation page and found to be incorrect for the current purchase. 

So, on second and subsequent trips to payment_confirmation, process_button will push nothing related to the use of in context, including if the user were to go back as far as the shopping cart and then use in context payment if then immediately delivered to the checkout_confirmation page.

So two options at that point seemed possible. Make it so that some or all of the session data was removed before getting to that point, or remove the check of the existing session data and lose the operational benefit of having/maintaining that data. From the notes in paypalwpp.php it appears that the goal is to maintain the session data, so the only thing that seemed viable was to remove the session data when it would appear as most likely to need to be rebuilt anyways (cancel). So, then the issue of the checkout.js file's involvement, well, if the session data is to prevent use of in context on the checkout_confirmation page, then it is possible that it was used in advance of that page but that page is/was to be shown and the lightbox removed. To accomplish the later part requires the checkout.js file somewhere on page, but with it inside of the form code, it never seemed to work. Once pulled out, it seemed to work though came across some issues with firefox and chrome. Chrome outright didn't like the iframe if it was hosted through https and the source site was not. Firefox was not as direct about the issue.

So, perhaps that answers the other side of the potential confusion. (not sure to which aspect the question was expected to cover)